src/formatDate.js

```diff
@@ -51,6 +51,54 @@
 
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromDayArray = date.split(fromFormat[3]);
+
+  const newObject = {};
+
+  newObject[`${fromFormat[0]}`] = fromDayArray[0];
+  newObject[`${fromFormat[1]}`] = fromDayArray[1];
+  newObject[`${fromFormat[2]}`] = fromDayArray[2];
+
+  const toDayArray = [];
+
+  for (let i = 0; i < 3; i++) {
+    for (const key in newObject) {
```

Comment:
It's better to do 2 separate loops (with their own `switch`) instead of nested loops